bindings/java/src/integration/com/apple/foundationdb/DirectoryTest.java

```diff
@@ -0,0 +1,233 @@
+/*
+ * DirectoryTest.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2018 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.apple.foundationdb;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CompletionException;
+
+import com.apple.foundationdb.directory.DirectoryAlreadyExistsException;
+import com.apple.foundationdb.directory.DirectoryLayer;
+import com.apple.foundationdb.directory.DirectorySubspace;
+import com.apple.foundationdb.directory.NoSuchDirectoryException;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+
+/**
+ * Integration tests for directory logic in FDB. This test requires a running FDB instance to work properly; 
+ * if one cannot be detected, all tests will be skipped.
+ */
+@ExtendWith(RequiresDatabase.class)
+public class DirectoryTest {
+    private static final FDB fdb = FDB.selectAPIVersion(700);
+
+    @Test
+    public void testCanCreateDirectory() throws Exception {
```

Comment:
```suggestion
    void testCanCreateDirectory() throws Exception {
```
Tests do not need to be public.

Reply:
Elsewhere as well.